src/Platform/Microsoft.Testing.Platform/ServerMode/JsonRpc/ServerModePerCallOutputDevice.cs

```diff
@@ -1,14 +1,44 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System.Collections.Concurrent;
+using System.Globalization;
+using System.Text;
+
 using Microsoft.Testing.Platform.Extensions.OutputDevice;
 using Microsoft.Testing.Platform.Helpers;
+using Microsoft.Testing.Platform.Hosts;
+using Microsoft.Testing.Platform.Logging;
 using Microsoft.Testing.Platform.OutputDevice;
+using Microsoft.Testing.Platform.Resources;
+using Microsoft.Testing.Platform.Services;
 
 namespace Microsoft.Testing.Platform.ServerMode;
 
-internal class ServerModePerCallOutputDevice : IPlatformOutputDevice
+internal sealed class ServerModePerCallOutputDevice : IPlatformOutputDevice
 {
+    private readonly IServiceProvider _serviceProvider;
+    private readonly ConcurrentBag<ServerLogMessage> _messages = new();
+
+    private IServerTestHost? _serverTestHost;
+
+    private static readonly string[] NewLineStrings = { "\r\n", "\n" };
+
+    public ServerModePerCallOutputDevice(IServiceProvider serviceProvider)
+        => _serviceProvider = serviceProvider;
+
+    internal async Task InitializeAsync(IServerTestHost serverTestHost)
```

Comment:
Could you add a small comment for future readers as to why we do that

Reply:
Added a comment. Does it look good?